torch/csrc/distributed/rpc/rref.cpp

```diff
@@ -133,14 +136,36 @@ template <>
 std::shared_ptr<ivalue::Future> UserRRef<IValue>::toHere() {
   auto future = std::make_shared<ivalue::Future>(nullptr);
   auto agent = RpcAgent::getDefaultRpcAgent();
-  auto futureResponse = agent->send(
+
+  // ScriptRRefFetchCall message always carries autograd context id even if
+  // the message itself does not contain any tensor, because the response would
+  // potentially contain tensors.
+  auto futureResponse = autograd::sendMessageWithAutograd(
+      *agent,
       agent->getWorkerInfo(ownerId_),
-      ScriptRRefFetchCall(rrefId()).toMessage());
+      ScriptRRefFetchCall(ownerId_, rrefId()).toMessage(),
+      false /* checkRequiresGrad */);
 
   futureResponse->addCallback([future](const Message& message) {
     RRefContext::handleException(message);
-    auto rfr = ScriptRRefFetchRet::fromMessage(message);
-    future->markCompleted(rfr->values().front());
+    if (message.type() == MessageType::FORWARD_AUTOGRAD_RESP) {
+      auto response = deserializeResponse(message);
```

Comment:
nit: This deserializeResponse can be outside the if statement right? Then the result can be used in either of the if-else clauses?